internal/dao/sqlx/reservation_dao.go

```diff
@@ -150,11 +171,28 @@ func (di *reservationDaoSqlx) CreateInstance(ctx context.Context, reservation *m
 	return nil
 }
 
+func (di *reservationDaoSqlx) GetByIdAllAccounts(ctx context.Context, id int64) (*models.Reservation, error) {
```

Comment:
This shall not be scoped by the account? We have the account for which the job was executed, or is there another reason to not scope by account?

Reply:
Right, I was rushing this and I forgot that I actually do have the account ID at the call site! So let's definitely add this for safety measures (someone else could accidentally use the function without account). Good call.

Reply:
So I did a full review of the whole Reservation DAO interface and I added scope to the two functions which take user's input. I fully documented everything:

```go
// ReservationDao represents a reservation, an abstraction of one or more background jobs with
// associated detail information different for different cloud providers (like number of vCPUs,
// instance IDs created etc).
type ReservationDao interface {
	// CreateNoop creates no operation reservation with details in a single transaction.
	CreateNoop(ctx context.Context, reservation *models.NoopReservation) error

	// CreateAWS creates AWS reservation with details in a single transaction.
	CreateAWS(ctx context.Context, reservation *models.AWSReservation) error

	// CreateInstance inserts instance associated to a reservation.
	CreateInstance(ctx context.Context, reservation *models.ReservationInstance) error

	// GetById returns reservation for a particular account.
	GetById(ctx context.Context, id int64) (*models.Reservation, error)

	// List returns reservation for a particular account.
	List(ctx context.Context, limit, offset int64) ([]*models.Reservation, error)

	// ListInstances returns instances associated to a reservation. UNSCOPED.
	ListInstances(ctx context.Context, limit, offset int64) ([]*models.ReservationInstance, error)

	// UpdateStatus sets status field and increment step counter by addSteps. UNSCOPED.
	UpdateStatus(ctx context.Context, id int64, status string, addSteps int32) error

	// UpdateReservationIDForAWS updates AWS reservation id field. UNSCOPED.
	UpdateReservationIDForAWS(ctx context.Context, id int64, awsReservationId string) error

	// FinishWithSuccess sets Success flag. UNSCOPED.
	FinishWithSuccess(ctx context.Context, id int64) error

	// FinishWithError sets Success flag and Error flag. UNSCOPED.
	FinishWithError(ctx context.Context, id int64, error string) error

	// Delete deletes a reservation. Only used in tests and background cleanup job. UNSCOPED.
	Delete(ctx context.Context, id int64) error
}
```

Reply:
FYI @avitova this PR will add few new test functions into the `reservation_test.go` from DAO test suite, I will do minimum possible changes to get it working, I will copy paste from existing functions. For the record, the new functions are:

* TestUpdateStepReservation
* TestFinishWithErrorReservation

No other changes, I know you are still working on this so hopefully this should help you with git rebasing.